### ISSUE TITLE
ICE after `cargo test --workspace`, not reproducible

### ISSUE THREAD

**Issue by Kinrany** [rust-lang/rust#118923]
I ran `cargo test --workspace` after an equivalent of a git-bisect session. Rust-analyzer was running in parallel.

Running `cargo test --workspace` for the second time did not result in the same error, so I can't really reproduce and provide a minimal code example, sorry.

### Error output

```
thread 'rustc' panicked at /rustc/6b771f6b5a6c8b03b6322a9c77ac77cb346148f0/compiler/rustc_middle/src/ty/context/tls.rs:111:50:
no ImplicitCtxt stored in tls
stack backtrace:
   0:        0x104e57dac - std::backtrace::Backtrace::create::h51e165066ce171e3
   1:        0x10e24aaa8 - <alloc[637795299c95d621]::boxed::Box<rustc_driver_impl[db4a640f955901f5]::install_ice_hook::{closure#0}> as core[339c3478672cf177]::ops::function::Fn<(&dyn for<'a, 'b> core[339c3478672cf177]::ops::function::Fn<(&'a core[339c3478672cf177]::panic::panic_info::PanicInfo<'b>,), Output = ()> + core[339c3478672cf177]::marker::Sync + core[339c3478672cf177]::marker::Send, &core[339c3478672cf177]::panic::panic_info::PanicInfo)>>::call
   2:        0x104e6fca8 - std::panicking::rust_panic_with_hook::hf562b6af24c16505
   3:        0x104e6fa70 - std::panicking::begin_panic_handler::{{closure}}::he0b4ebe231153083
   4:        0x104e6d25c - std::sys_common::backtrace::__rust_end_short_backtrace::h506e293342848289
   5:        0x104e6f80c - _rust_begin_unwind
   6:        0x104ec9108 - core::panicking::panic_fmt::hdbf482c928a0b9a2
   7:        0x104ec90d4 - core::option::expect_failed::h02d56d537e4d16e7
   8:        0x10e2188cc - rustc_interface[85513576697f58d]::util::run_in_thread_pool_with_globals::<rustc_interface[85513576697f58d]::interface::run_compiler<core[339c3478672cf177]::result::Result<(), rustc_span[bf241269b6f97e85]::ErrorGuaranteed>, rustc_driver_impl[db4a640f955901f5]::run_compiler::{closure#1}>::{closure#0}, core[339c3478672cf177]::result::Result<(), rustc_span[bf241269b6f97e85]::ErrorGuaranteed>>::{closure#2}
   9:        0x111f7c438 - <rayon_core[d9bbcca4c17cfd59]::sleep::Sleep>::sleep
  10:        0x111f7b940 - <rayon_core[d9bbcca4c17cfd59]::registry::WorkerThread>::wait_until_cold
  11:        0x10d96a5c8 - <rayon_core[d9bbcca4c17cfd59]::registry::ThreadBuilder>::run
  12:        0x10e25fbb0 - rustc_span[bf241269b6f97e85]::set_session_globals_then::<(), rustc_interface[85513576697f58d]::util::run_in_thread_pool_with_globals<rustc_interface[85513576697f58d]::interface::run_compiler<core[339c3478672cf177]::result::Result<(), rustc_span[bf241269b6f97e85]::ErrorGuaranteed>, rustc_driver_impl[db4a640f955901f5]::run_compiler::{closure#1}>::{closure#0}, core[339c3478672cf177]::result::Result<(), rustc_span[bf241269b6f97e85]::ErrorGuaranteed>>::{closure#3}::{closure#0}::{closure#0}::{closure#0}>
  13:        0x10e218c48 - <<crossbeam_utils[a385c9f1894b556d]::thread::ScopedThreadBuilder>::spawn<<rayon_core[d9bbcca4c17cfd59]::ThreadPoolBuilder>::build_scoped<rustc_interface[85513576697f58d]::util::run_in_thread_pool_with_globals<rustc_interface[85513576697f58d]::interface::run_compiler<core[339c3478672cf177]::result::Result<(), rustc_span[bf241269b6f97e85]::ErrorGuaranteed>, rustc_driver_impl[db4a640f955901f5]::run_compiler::{closure#1}>::{closure#0}, core[339c3478672cf177]::result::Result<(), rustc_span[bf241269b6f97e85]::ErrorGuaranteed>>::{closure#3}::{closure#0}::{closure#0}, rustc_interface[85513576697f58d]::util::run_in_thread_pool_with_globals<rustc_interface[85513576697f58d]::interface::run_compiler<core[339c3478672cf177]::result::Result<(), rustc_span[bf241269b6f97e85]::ErrorGuaranteed>, rustc_driver_impl[db4a640f955901f5]::run_compiler::{closure#1}>::{closure#0}, core[339c3478672cf177]::result::Result<(), rustc_span[bf241269b6f97e85]::ErrorGuaranteed>>::{closure#3}::{closure#0}::{closure#1}, core[339c3478672cf177]::result::Result<(), rustc_span[bf241269b6f97e85]::ErrorGuaranteed>>::{closure#0}::{closure#0}::{closure#0}, ()>::{closure#0} as core[339c3478672cf177]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  14:        0x10e266e48 - std[d82c2182fc4caff7]::sys_common::backtrace::__rust_begin_short_backtrace::<alloc[637795299c95d621]::boxed::Box<dyn core[339c3478672cf177]::ops::function::FnOnce<(), Output = ()> + core[339c3478672cf177]::marker::Send>, ()>
  15:        0x10e239b0c - <<std[d82c2182fc4caff7]::thread::Builder>::spawn_unchecked_<alloc[637795299c95d621]::boxed::Box<dyn core[339c3478672cf177]::ops::function::FnOnce<(), Output = ()> + core[339c3478672cf177]::marker::Send>, ()>::{closure#1} as core[339c3478672cf177]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  16:        0x104e77ea4 - std::sys::unix::thread::Thread::new::thread_start::h00dd4ee37dcd6afc
  17:        0x19defffa8 - __pthread_joiner_wake


rustc version: 1.76.0-nightly (6b771f6b5 2023-11-15)
platform: aarch64-apple-darwin
```

**Comment by petrochenkov** [comment]
Duplicate of https://github.com/rust-lang/rust/issues/111522.
